src/components/Wizard/Wizard.test.tsx

```diff
@@ -126,28 +163,88 @@ describe('<Wizard />', () => {
 		},
 	);
 
-	test('Should not submit when invalid data', () => {
+	test('should go to navigate to previous step when clicking previousbutton', () => {
+		const pushSpy = jest.fn();
+		(useHistory as jest.Mock).mockReturnValue({
+			push: pushSpy,
+		});
 		renderComponent(
 			{
 				file: mockFile,
-				metadata: { documentDescription: '', executionDate: '' },
+				metadata: { mockData },
 			},
 			'/step2',
 		);
-		fireEvent.click(screen.getByText('Opslaan'));
-		expect(onMetadataSubmitMock).not.toHaveBeenCalled();
+		const previousButton = screen.getByTestId('previous-button');
+		expect(screen.getByTestId('previous-button')).toBeInTheDocument();
+		fireEvent.click(previousButton);
+
+		expect(pushSpy).toHaveBeenCalledWith('/');
 	});
+	describe('onMetadataSubmit()', () => {
+		test('is called when saving while <MetaDataForm /> is valid', async () => {
+			const pushSpy = jest.fn();
+			const setMetadataSpy = jest.spyOn(actions, 'setMetadata');
+			useHistoryMock.mockReturnValue({ push: pushSpy } as any);
+			const metadata: MetadataExample = {
+				documentDescription: '__DOCUMENT_DESCRIPTION__',
+				executionDate: '__EXECUTION_DATE__',
+			};
+			getMetadataFromStoreMock.mockReturnValue(metadata);
+			renderComponent({ file: mockFile, metadata: {} }, '/step2');
+			const { onChange } = mockComponentProps<ComponentProps<typeof MetadataForm>>(MetadataFormMock);
+			act(() => {
+				onChange(metadata, true, []);
+			});
+			act(() => {
+				fireEvent.click(screen.getByText('Opslaan'));
+			});
+			await waitFor(() => {
+				expect(setMetadataSpy).toHaveBeenCalledWith(metadata);
+				expect(onMetadataSubmitMock).toHaveBeenCalledWith({
+					file: mockFile,
+					metadata,
+				});
+				expect(pushSpy).toHaveBeenCalledWith('/');
+			});
+		});
+
+		test('is NOT called when saving while <MetaDataForm /> is NOT valid', async () => {
+			const pushSpy = jest.fn();
+			useHistoryMock.mockReturnValue({ push: pushSpy } as any);
+			renderComponent({ file: mockFile, metadata: {} }, '/step2');
+
+			act(() => {
```

Comment:
Would it not be more efficient here to check if the button has a disabled attribute?